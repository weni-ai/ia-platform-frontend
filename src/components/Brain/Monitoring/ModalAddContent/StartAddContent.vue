<template>
  <section class="add-content">
    <UnnnicIcon
      class="add-content__icon"
      :icon="icon"
      :filled="iconFilled"
      size="xl"
      scheme="neutral-soft"
      data-testid="icon"
    />
    <section class="add-content__infos">
      <UnnnicIntelligenceText
        color="neutral-cloudy"
        family="secondary"
        size="body-gt"
        tag="p"
        data-testid="description"
      >
        {{ description }}
      </UnnnicIntelligenceText>
      <UnnnicIntelligenceText
        class="text-sub-description"
        color="neutral-clean"
        family="secondary"
        size="body-md"
        tag="p"
        data-testid="sub-description"
      >
        {{ subDescription }}
      </UnnnicIntelligenceText>
    </section>

    <UnnnicButton
      class="add-content__trigger"
      type="secondary"
      iconLeft="add"
      data-testid="add-button"
      @click="$emit('add')"
    >
      {{ textAddContent }}
    </UnnnicButton>
  </section>
</template>

<script setup>
const props = defineProps({
  description: {
    type: String,
    default: '',
  },
  subDescription: {
    type: String,
    default: '',
  },
  icon: {
    type: String,
    default: 'text_snippet',
  },
  iconFilled: {
    type: Boolean,
    default: true,
  },
  textAddContent: {
    type: String,
    default: '',
  },
});

const emit = defineEmits(['add']);
</script>

<style scoped lang="scss">
.add-content {
  display: grid;
  justify-items: center;
  align-content: center;
  &__icon {
    margin-bottom: $unnnic-spacing-xs;
  }

  &__infos {
    display: grid;
    justify-items: center;
  }

  &__trigger {
    margin-top: $unnnic-spacing-sm;
  }
}
</style>
