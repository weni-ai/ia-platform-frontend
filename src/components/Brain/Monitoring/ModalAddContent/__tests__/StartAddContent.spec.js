import { mount } from '@vue/test-utils';
import { describe, it, expect, vi, beforeEach } from 'vitest';

import StartAddContent from '../StartAddContent.vue';

describe('StartAddContent.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(StartAddContent, {
      props: {
        icon: 'custom-icon',
        iconFilled: false,
        description: 'This is a description',
        subDescription: 'This is a sub-description',
        textAddContent: 'Add Item',
      },
    });
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders the icon with correct props', () => {
    const icon = wrapper.findComponent('[data-testid="icon"]');
    expect(icon.exists()).toBe(true);
    expect(icon.props('icon')).toBe('custom-icon');
    expect(icon.props('filled')).toBe(false);
  });

  it('renders description and subDescription texts', () => {
    const description = wrapper.find('[data-testid="description"]');
    const subDescription = wrapper.find('[data-testid="sub-description"]');
    expect(description.exists()).toBe(true);
    expect(description.text()).toBe('This is a description');
    expect(subDescription.exists()).toBe(true);
    expect(subDescription.text()).toBe('This is a sub-description');
  });

  it('renders the add content button with correct text', () => {
    const button = wrapper.findComponent('[data-testid="add-button"]');
    expect(button.exists()).toBe(true);
    expect(button.text()).toContain('Add Item');
  });

  it('emits the "add" event when button is clicked', async () => {
    const button = wrapper.findComponent('[data-testid="add-button"]');
    await button.trigger('click');

    expect(wrapper.emitted('add')).toBeTruthy();
    expect(wrapper.emitted('add').length).toBe(1);
  });
});
