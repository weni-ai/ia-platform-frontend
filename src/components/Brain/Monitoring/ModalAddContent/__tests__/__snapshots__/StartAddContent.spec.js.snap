// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StartAddContent.vue > matches the snapshot 1`] = `
"<section data-v-c26a5569="" class="add-content"><span data-v-76ceda5d="" data-v-c26a5569="" class="material-symbols-rounded unnnic-icon-scheme--neutral-soft unnnic-icon-size--xl add-content__icon" data-testid="icon" translate="no">custom-icon</span>
  <section data-v-c26a5569="" class="add-content__infos">
    <p data-v-ec8da4ad="" data-v-c26a5569="" class="unnnic-text unnnic-text--size-body-gt unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-regular unnnic-text--color-neutral-cloudy" data-testid="description">This is a description</p>
    <p data-v-ec8da4ad="" data-v-c26a5569="" class="unnnic-text unnnic-text--size-body-md unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-regular unnnic-text--color-neutral-clean text-sub-description text-sub-description" data-testid="sub-description">This is a sub-description</p>
  </section><button data-v-ed84dbab="" data-v-c26a5569="" class="add-content__trigger unnnic-button unnnic-button--size-large unnnic-button--secondary add-content__trigger" data-testid="add-button">
    <!----><span data-v-76ceda5d="" data-v-ed84dbab="" class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--md unnnic-button__icon-left" data-testid="icon-left" translate="no">add</span>
    <!----><span data-v-ed84dbab="" class="unnnic-button__label" data-testid="button-label">Add Item </span>
    <!---->
  </button>
</section>"
`;
